### ISSUE TITLE
unkillable crash (mkv edition, switching streams) (dup)

### ISSUE THREAD

**Issue by eladkarako** [stax76/mpv.net#450]
TL;DR - video attached below.

<h3>input</h3>

I've generated a mkv file from http://www.youtube.com/watch?v=4o3q5Yvu0mg,  
https://github.com/eladkarako/yt-dlp_kit,  
and `yt-dlp_conf_variation_audio_only__prefer_m4a__no_re_encoding_just_mux__mkv_container.cmd`

the file has audio only stream, couple of generated subtitles and a text description, and a thumbnail (cover.jpg) soft-embedded into a mkv container.

```txt
Audio: AAC 44100Hz stereo [A: ISO Media file produced by Google Inc. [eng] (aac lc, 44100 Hz, stereo) [default]]
Subtitle: WebVTT [S: English [eng] (webvtt) [default]]
Subtitle: WebVTT [S: Hebrew [heb] (webvtt)]
Subtitle: WebVTT [S: Japanese [jpn] (webvtt)]
Subtitle: WebVTT [S: Korean [kor] (webvtt)]
Subtitle: UTF-8 [S: No subtitles]
```

<details><summary>more stream details</summary>

```txt
General
Unique ID                      : 83420247810375678697352180401008551937 (0x3EC227C33927EB66E10A47CC74FC2801)
Complete name                  : D:\Documents\Videos\calm\Music - Japanese Lofi Mix - Yor Forger [4o3q5Yvu0mg].mkv
Format                         : Matroska
Format version                 : Version 4
File size                      : 175 MiB
Duration                       : 3 h 7 min
Overall bit rate               : 131 kb/s
Track name                     : Yor Forger ヨル・フォージャー ☯ Japanese Lofi HipHop Mix
Description                    : 🎧Japanese Spotify Playlist: https://open.spotify.com/playlist/3Qt7TvEut6lKzZfhiTKAQA?si=f083990f52c24b28&fbclid=IwAR0llCUsSFAn9HdlJjAXLjynM-7NGJnXquaJXAPTMfB3YT1ghd0dMO9hHUE&nd=1 /  / 🎧Chill Japanese Songs Playlist: https://open.spotify.com/playlist/2AxATrsYnX6YeB7rGKNKks?fbclid=IwAR3hXEG6Q7Jn1MrRFtyiY0xzJBBgJ7oD9q-Vf8xfJxa6sS_q66X-4aCqe80 /  / 🎧🎧NEW EP ON Spotify: https://open.spotify.com/album/3kfrhfFisK4zoAAUkWTKRa /  / ------------------------------------------------------------------------------------------------------------------- / Please support the artists by buying the original tracks. / Enjoy and please subscribe to the channel :) / -------------------------------------------------------------------------------------------------------------------- / Image: https://www.artstation.com/artwork/2q53LB / Artist: CISL / ArtStation: https://cisl.artstation.com/ / Linktree: https://linktr.ee/CISL / Instagram: https://www.instagram.com/ci.s.l/ / Gumroad: https://cisl.gumroad.com/ / Youtube: https://www.youtube.com/channel/UCZzjKB9pMRYSm3lkuOC6SPQ / Gmail: tea.createa@gmail.com /  / ------------------------------------------------------------------------------------------------------------------ / spy x family /  / Tracklist:  / 00:00 - Vindu - Land Of The Red Dragon / 03:13 - Vindu - Yamadera / 07:14 - Deebu - Bittersweet / 09:18 - Deebu - Kanai / 11:12 - DaniSogen - Kagayaku / 13:23 - Tenno - Midnight Story / 15:30 - Vindu - Senso / 18:40 - Raimu - Dozing Below The Trees / 20:45 - Deebu -Lotus Book / 23:13 - greenface - Haruka / 26:00 - DaniSogen - Waiting For Yoaku / 28:11 - Nogymx - Harbor at Moonrise / 30:34 - Deebu - Ukiyo / 32:41 - Nogymx - Clarity / 34:57 - Raimu - Jama Shinaide Kudasai / 37:30 - Sukonjura - Ukiyo / 39:46 - Vindu - Streets of Kyoto / 42:13 - Vindu - Red On White / 44:50 - Vindu - Nara Blues / 48:02 - Tenno - Yamabiko / 50:19 - Sukonjura - Yukisora / 52:35 - Raimu & Vindu - Entering The Golden Gai / 55:30 - Tenno - Muramasa / 57:51 - Nogymx & DaniSogen - Nocturne /  / Tracks by: Raimu, DaniSogen, Tenno, Vindu, Deebu, Sukonjura, Nogymx,  greenface /  / Follow DaniSogen: / 🎧 Spotify: https://open.spotify.com/artist/7igN8... / YouTube: https://www.youtube.com/channel/UCb3zte64T8YRvq9iN4WxQaA / Instagram: https://www.instagram.com/danisogen/ / Soundcloud: https://soundcloud.com/danisogen /  / Follow Raimu: / 🎧 Spotify: https://spoti.fi/2S0psGT / YouTube: https://bit.ly/2sZ8bFq / Instagram: https://instagram.com/raimu.music / Soundcloud: https://soundcloud.com/raimu_music /  / Follow Tenno: / ➜ https://open.spotify.com/artist/3yu4V... / ➜ https://soundcloud.com/musictenno / ➜ https://www.instagram.com/tennomusic/ / ➜ https://twitter.com/tennomusic / 🎧 Stream / Download / https://linktr.ee/tenno /  / Follow Vindu:  / ➜ https://soundcloud.com/vindumusic / ➜ https://www.facebook.com/vindumusic/ / ➜ https://twitter.com/vindumusic/ / ➜ https://www.instagram.com/vindumusic/ / 🎧 https://open.spotify.com/artist/2CRx4wZoJ1wJLX6DtKJpej / ➜https://vindumusic.bandcamp.com/ /  / Follow Deebu: / ➜Spotify: https://open.spotify.com/artist/6OZJtCCwoj5Ya0NudLT4xG?si=KFX8ltlORcOp8hA5M3Vfqw&dl_branch=1&nd=1 / ➜Twitter: https://twitter.com/deebumusic / 🎧Stores: https://distrokid.com/hyperfollow/deebu/wind-of-fate /  / Follow Nogymx: / 🎧 Spotify - https://spoti.fi/32PDLnQ / Instagram - https://www.instagram.com/nogymx / Soundcloud - https://soundcloud.com/nogymx / Twitter - https://www.twitter.com/nogymx / Bandcamp - http://nogymx.bandcamp.com /  / Follow Sukonjura:  / ➜SoundCloud: https://soundcloud.com/sukonjura / ➜Instagram: https://www.instagram.com/sukonjura/?hl=en / ➜Spotify: https://open.spotify.com/artist/36guDPm0XWdxKKgvoOA9VP?si=PFQ_pgxZRuimfPGjVy-r3Q / ➜Itunes: https://itunes.apple.com/ca/artist/sukonjura/id1440715933 / ➜Pandora: https://www.pandora.com/restricted / ➜Gmail: sukonjura@gmail.com /  / Follow greenface: / ➜https://www.instagram.com/greenmid2k/ / ➜https://www.youtube.com/channel/UCnjfdZmjF0EcYGWlTIKPM6g / ➜https://open.spotify.com/artist/2aTEuIyczpLxQw9I0UiEFL?si=LQYn1JbIRXKPGJ4eO5JWZQ&nd=1 / ➜https://music.apple.com/jp/artist/greenface/1469637411 /  / ------------------------------------------------------------------------------------------------------------------- / Please support the artists by buying the original tracks. / Enjoy and please subscribe to the channel :) / -------------------------------------------------------------------------------------------------------------------- /  / #lofi #lofihiphop #studymix /  / lofi hip hop mix,jazzhop mix,lofi study music,lofi chill mix,lofi hip hop beats,lofi chill beats,chillhop mix,study chill beats,lofi study mix,chilledcow,lofi hip hop,lofi,lofi hiphop,mix,lo fi,hiphop,lo-fi,lo-fi mix,lo fi mix,lo fi music,lofi mix,chill mix,hip,pt. 1,pt.2,pt.3,pt.4hop,vibe,instrumental,beat,beats,meditation,sleep,relax,music,lo fi hip-hop,study music,relaxing music,lofi remix,old songs,old lo fi songs,lofi old songs,old songs but it's lofi remix
Writing application            : Lavf59.25.100
Writing library                : Lavf59.25.100
Cover                          : Yes
Comment                        : http://www.youtube.com/watch?v=4o3q5Yvu0mg
ErrorDetectionType             : Per level 1
Attachments                    : cover.jpg
ARTIST                         : Vindu
DATE                           : 20220502
PURL                           : http://www.youtube.com/watch?v=4o3q5Yvu0mg
SYNOPSIS                       : 🎧Japanese Spotify Playlist: https://open.spotify.com/playlist/3Qt7TvEut6lKzZfhiTKAQA?si=f083990f52c24b28&fbclid=IwAR0llCUsSFAn9HdlJjAXLjynM-7NGJnXquaJXAPTMfB3YT1ghd0dMO9hHUE&nd=1 /  / 🎧Chill Japanese Songs Playlist: https://open.spotify.com/playlist/2AxATrsYnX6YeB7rGKNKks?fbclid=IwAR3hXEG6Q7Jn1MrRFtyiY0xzJBBgJ7oD9q-Vf8xfJxa6sS_q66X-4aCqe80 /  / 🎧🎧NEW EP ON Spotify: https://open.spotify.com/album/3kfrhfFisK4zoAAUkWTKRa /  / ------------------------------------------------------------------------------------------------------------------- / Please support the artists by buying the original tracks. / Enjoy and please subscribe to the channel :) / -------------------------------------------------------------------------------------------------------------------- / Image: https://www.artstation.com/artwork/2q53LB / Artist: CISL / ArtStation: https://cisl.artstation.com/ / Linktree: https://linktr.ee/CISL / Instagram: https://www.instagram.com/ci.s.l/ / Gumroad: https://cisl.gumroad.com/ / Youtube: https://www.youtube.com/channel/UCZzjKB9pMRYSm3lkuOC6SPQ / Gmail: tea.createa@gmail.com /  / ------------------------------------------------------------------------------------------------------------------ / spy x family /  / Tracklist:  / 00:00 - Vindu - Land Of The Red Dragon / 03:13 - Vindu - Yamadera / 07:14 - Deebu - Bittersweet / 09:18 - Deebu - Kanai / 11:12 - DaniSogen - Kagayaku / 13:23 - Tenno - Midnight Story / 15:30 - Vindu - Senso / 18:40 - Raimu - Dozing Below The Trees / 20:45 - Deebu -Lotus Book / 23:13 - greenface - Haruka / 26:00 - DaniSogen - Waiting For Yoaku / 28:11 - Nogymx - Harbor at Moonrise / 30:34 - Deebu - Ukiyo / 32:41 - Nogymx - Clarity / 34:57 - Raimu - Jama Shinaide Kudasai / 37:30 - Sukonjura - Ukiyo / 39:46 - Vindu - Streets of Kyoto / 42:13 - Vindu - Red On White / 44:50 - Vindu - Nara Blues / 48:02 - Tenno - Yamabiko / 50:19 - Sukonjura - Yukisora / 52:35 - Raimu & Vindu - Entering The Golden Gai / 55:30 - Tenno - Muramasa / 57:51 - Nogymx & DaniSogen - Nocturne /  / Tracks by: Raimu, DaniSogen, Tenno, Vindu, Deebu, Sukonjura, Nogymx,  greenface /  / Follow DaniSogen: / 🎧 Spotify: https://open.spotify.com/artist/7igN8... / YouTube: https://www.youtube.com/channel/UCb3zte64T8YRvq9iN4WxQaA / Instagram: https://www.instagram.com/danisogen/ / Soundcloud: https://soundcloud.com/danisogen /  / Follow Raimu: / 🎧 Spotify: https://spoti.fi/2S0psGT / YouTube: https://bit.ly/2sZ8bFq / Instagram: https://instagram.com/raimu.music / Soundcloud: https://soundcloud.com/raimu_music /  / Follow Tenno: / ➜ https://open.spotify.com/artist/3yu4V... / ➜ https://soundcloud.com/musictenno / ➜ https://www.instagram.com/tennomusic/ / ➜ https://twitter.com/tennomusic / 🎧 Stream / Download / https://linktr.ee/tenno /  / Follow Vindu:  / ➜ https://soundcloud.com/vindumusic / ➜ https://www.facebook.com/vindumusic/ / ➜ https://twitter.com/vindumusic/ / ➜ https://www.instagram.com/vindumusic/ / 🎧 https://open.spotify.com/artist/2CRx4wZoJ1wJLX6DtKJpej / ➜https://vindumusic.bandcamp.com/ /  / Follow Deebu: / ➜Spotify: https://open.spotify.com/artist/6OZJtCCwoj5Ya0NudLT4xG?si=KFX8ltlORcOp8hA5M3Vfqw&dl_branch=1&nd=1 / ➜Twitter: https://twitter.com/deebumusic / 🎧Stores: https://distrokid.com/hyperfollow/deebu/wind-of-fate /  / Follow Nogymx: / 🎧 Spotify - https://spoti.fi/32PDLnQ / Instagram - https://www.instagram.com/nogymx / Soundcloud - https://soundcloud.com/nogymx / Twitter - https://www.twitter.com/nogymx / Bandcamp - http://nogymx.bandcamp.com /  / Follow Sukonjura:  / ➜SoundCloud: https://soundcloud.com/sukonjura / ➜Instagram: https://www.instagram.com/sukonjura/?hl=en / ➜Spotify: https://open.spotify.com/artist/36guDPm0XWdxKKgvoOA9VP?si=PFQ_pgxZRuimfPGjVy-r3Q / ➜Itunes: https://itunes.apple.com/ca/artist/sukonjura/id1440715933 / ➜Pandora: https://www.pandora.com/restricted / ➜Gmail: sukonjura@gmail.com /  / Follow greenface: / ➜https://www.instagram.com/greenmid2k/ / ➜https://www.youtube.com/channel/UCnjfdZmjF0EcYGWlTIKPM6g / ➜https://open.spotify.com/artist/2aTEuIyczpLxQw9I0UiEFL?si=LQYn1JbIRXKPGJ4eO5JWZQ&nd=1 / ➜https://music.apple.com/jp/artist/greenface/1469637411 /  / ------------------------------------------------------------------------------------------------------------------- / Please support the artists by buying the original tracks. / Enjoy and please subscribe to the channel :) / -------------------------------------------------------------------------------------------------------------------- /  / #lofi #lofihiphop #studymix /  / lofi hip hop mix,jazzhop mix,lofi study music,lofi chill mix,lofi hip hop beats,lofi chill beats,chillhop mix,study chill beats,lofi study mix,chilledcow,lofi hip hop,lofi,lofi hiphop,mix,lo fi,hiphop,lo-fi,lo-fi mix,lo fi mix,lo fi music,lofi mix,chill mix,hip,pt. 1,pt.2,pt.3,pt.4hop,vibe,instrumental,beat,beats,meditation,sleep,relax,music,lo fi hip-hop,study music,relaxing music,lofi remix,old songs,old lo fi songs,lofi old songs,old songs but it's lofi remix

Audio
ID                             : 1
Format                         : AAC LC
Format/Info                    : Advanced Audio Codec Low Complexity
Codec ID                       : A_AAC-2
Duration                       : 3 h 7 min
Channel(s)                     : 2 channels
Channel layout                 : L R
Sampling rate                  : 44.1 kHz
Frame rate                     : 43.066 FPS (1024 SPF)
Compression mode               : Lossy
Title                          : ISO Media file produced by Google Inc.
Language                       : English
Default                        : Yes
Forced                         : No
VENDOR_ID                      : [0][0][0][0]

Text #1
ID                             : 2
Format                         : D_WEBVTT/SUBTITLES
Codec ID                       : D_WEBVTT/SUBTITLES
Duration                       : 3 h 7 min
Title                          : English / English
Language                       : English
Default                        : Yes
Forced                         : No

Text #2
ID                             : 3
Format                         : D_WEBVTT/SUBTITLES
Codec ID                       : D_WEBVTT/SUBTITLES
Duration                       : 3 h 7 min
Title                          : Hebrew / Hebrew
Language                       : Hebrew
Default                        : No
Forced                         : No

Text #3
ID                             : 4
Format                         : D_WEBVTT/SUBTITLES
Codec ID                       : D_WEBVTT/SUBTITLES
Duration                       : 3 h 7 min
Title                          : Japanese / Japanese
Language                       : Japanese
Default                        : No
Forced                         : No

Text #4
ID                             : 5
Format                         : D_WEBVTT/SUBTITLES
Codec ID                       : D_WEBVTT/SUBTITLES
Duration                       : 3 h 7 min
Title                          : Korean / Korean
Language                       : Korean
Default                        : No
Forced                         : No

Menu
00:00:00.000                   : :Vindu - Land Of The Red Dragon
00:03:13.000                   : :Vindu - Yamadera
00:07:14.000                   : :Deebu - Bittersweet
00:09:18.000                   : :Deebu - Kanai
00:11:12.000                   : :DaniSogen - Kagayaku
00:13:23.000                   : :Tenno - Midnight Story
00:15:30.000                   : :Vindu - Senso
00:18:40.000                   : :Raimu - Dozing Below The Trees
00:20:45.000                   : :Deebu -Lotus Book
00:23:13.000                   : :greenface - Haruka
00:26:00.000                   : :DaniSogen - Waiting For Yoaku
00:28:11.000                   : :Nogymx - Harbor at Moonrise
00:30:34.000                   : :Deebu - Ukiyo
00:32:41.000                   : :Nogymx - Clarity
00:34:57.000                   : :Raimu - Jama Shinaide Kudasai
00:37:30.000                   : :Sukonjura - Ukiyo
00:39:46.000                   : :Vindu - Streets of Kyoto
00:42:13.000                   : :Vindu - Red On White
00:44:50.000                   : :Vindu - Nara Blues
00:48:02.000                   : :Tenno - Yamabiko
00:50:19.000                   : :Sukonjura - Yukisora
00:52:35.000                   : :Raimu & Vindu - Entering The Golden Gai
00:55:30.000                   : :Tenno - Muramasa
00:57:51.000                   : :Nogymx & DaniSogen - Nocturne

```

</details>

<hr/>

<h3>issue</h3>

I seems to found a glitch of a kind,  
I can't really explain what is going on..
but the program halts in a way I can't even close it and have to use the task manager for that, it does not look like there is a heavy task running or anything, just the UI is locked up for some reason (if it was CPU heavy the OS would trigger the close probably..).

<h3>instructions</h3>
(video provided).

https://user-images.githubusercontent.com/415238/175820353-4c453f41-57cc-4ab5-b1fa-de4114c298a4.mp4

**Comment by dyphire** [comment]
dup https://github.com/stax76/mpv.net/issues/421#issuecomment-1132279304

**Comment by eladkarako** [comment]
thanks I wasn't aware of https://www.matroska.org/technical/chapters.html#edition-flags  
(https://www.matroska.org/files/matroska_file_format_alexander_noe.pdf#page=31)

I wonder if the fault is the way ffmpeg is using it,  
or yt-dlp not implementing something...

**Comment by eladkarako** [comment]
@stax76 I've reopened this to update,  
that I've updated `MediaInfo.dll` (to the currently available x64 dll from https://mediaarea.net/en/MediaInfo/Download/Windows)  
and `mpv-2.dll` (latest nightly 64bit from https://sourceforge.net/projects/mpv-player-windows/files/libmpv/)  

<img width="540" alt="2022-06-28_043145" src="https://user-images.githubusercontent.com/415238/176067612-678f6027-2612-4114-95bf-1bbd1e6970ff.png">

which seems to solve the issue for some reason (no errors).

https://user-images.githubusercontent.com/415238/176068158-9b0db726-6426-4c2b-9885-d61861de29ac.mp4
